engine/access/state_stream/backend/backend_executiondata.go

```diff
@@ -34,18 +34,20 @@ type ExecutionDataBackend struct {
 func (b *ExecutionDataBackend) GetExecutionDataByBlockID(ctx context.Context, blockID flow.Identifier) (*execution_data.BlockExecutionData, error) {
 	header, err := b.headers.ByBlockID(blockID)
 	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			return nil, fmt.Errorf("could not get block header for %s: %w", blockID, subscription.ErrBlockNotReady)
```

Comment:
we don't need to wrap the errors in this method since it's not a streaming endpoint.